src/display/api.js

```diff
@@ -850,385 +850,395 @@ class PDFDocumentProxy {
+      intentState.renderTasks.push(opListTask);
+      intentState.operatorList = {
+        fnArray: [],
+        argsArray: [],
+        lastChunk: false,
+      };
 
-    /**
-     * @param {getTextContentParameters} params - getTextContent parameters.
-     * @return {ReadableStream} ReadableStream to read textContent chunks.
-     */
-    streamTextContent(params = {}) {
-      const TEXT_CONTENT_CHUNK_SIZE = 100;
-      return this.transport.messageHandler.sendWithStream('GetTextContent', {
-        pageIndex: this.pageNumber - 1,
-        normalizeWhitespace: (params.normalizeWhitespace === true),
-        combineTextItems: (params.disableCombineTextItems !== true),
-      }, {
-        highWaterMark: TEXT_CONTENT_CHUNK_SIZE,
-        size(textContent) {
-          return textContent.items.length;
-        },
+      this._stats.time('Page Request');
+      this._transport.messageHandler.send('RenderPageRequest', {
+        pageIndex: this.pageIndex,
+        intent: renderingIntent,
       });
-    },
+    }
+    return intentState.opListReadCapability.promise;
+  }
 
-    /**
-     * @param {getTextContentParameters} params - getTextContent parameters.
-     * @return {Promise} That is resolved a {@link TextContent}
-     * object that represent the page text content.
-     */
-    getTextContent: function PDFPageProxy_getTextContent(params) {
-      params = params || {};
-      let readableStream = this.streamTextContent(params);
-
-      return new Promise(function(resolve, reject) {
-        function pump() {
-          reader.read().then(function({ value, done, }) {
-            if (done) {
-              resolve(textContent);
-              return;
-            }
-            Object.assign(textContent.styles, value.styles);
-            textContent.items.push(...value.items);
-            pump();
-          }, reject);
-        }
+  /**
+   * @param {getTextContentParameters} params - getTextContent parameters.
+   * @return {ReadableStream} ReadableStream to read textContent chunks.
+   */
+  streamTextContent({ normalizeWhitespace = false,
+                      disableCombineTextItems = false, } = {}) {
+    const TEXT_CONTENT_CHUNK_SIZE = 100;
+
+    return this._transport.messageHandler.sendWithStream('GetTextContent', {
+      pageIndex: this.pageNumber - 1,
+      normalizeWhitespace: normalizeWhitespace === true,
+      combineTextItems: disableCombineTextItems !== true,
+    }, {
+      highWaterMark: TEXT_CONTENT_CHUNK_SIZE,
+      size(textContent) {
+        return textContent.items.length;
+      },
+    });
+  }
 
-        let reader = readableStream.getReader();
-        let textContent = {
-          items: [],
-          styles: Object.create(null),
-        };
+  /**
+   * @param {getTextContentParameters} params - getTextContent parameters.
+   * @return {Promise} That is resolved a {@link TextContent}
+   * object that represent the page text content.
+   */
+  getTextContent(params) {
```

Comment:
Shouldn't this be `params = {}` to match the removed `params = params || {};` line?

Reply:
Technically it doesn't really matter, since `getTextContent` is basically just a convenient wrapper around `streamTextContent` now. However, given that the parameters are optional, it cannot hurt to indicate that with `params = {}` though :-)